File: java/shared/java/org/signal/libsignal/zkgroup/groups/GroupIdentifier.java
//
// Copyright 2020-2021 Signal Messenger, LLC.
// SPDX-License-Identifier: AGPL-3.0-only
//

package org.signal.libsignal.zkgroup.groups;

import org.signal.libsignal.zkgroup.InvalidInputException;
import org.signal.libsignal.zkgroup.internal.ByteArray;

public final class GroupIdentifier extends ByteArray {

  public static final int SIZE = 32;

  public GroupIdentifier(byte[] contents) throws InvalidInputException {
    super(contents, SIZE);
  }

}

File: swift/Sources/LibSignalClient/zkgroup/GroupIdentifier.swift
//
// Copyright 2020-2021 Signal Messenger, LLC.
// SPDX-License-Identifier: AGPL-3.0-only
//

import Foundation
import SignalFfi

public class GroupIdentifier: HashableByteArray, CustomStringConvertible, @unchecked Sendable {
    public static let SIZE: Int = 32

    public required init(contents: Data) throws {
        try super.init(newContents: contents, expectedLength: GroupIdentifier.SIZE)
    }

    /// Returns the group ID as (lowercase) hexadecimal
    public var description: String {
        self.serialize().toHex()
    }
}

private func witnessGroupIdentifierIsHashable() throws {
    func isHashable<T: Hashable>(_: T) {}
    isHashable(try GroupIdentifier(contents: Data()))
}
